app/controllers/halls_controller.rb

```diff
@@ -0,0 +1,52 @@
+class HallsController < ApplicationController
+
+    before_action :set_hall, only: [:show, :update, :destroy]
+
+    def index
+        @halls = Hall.all.map do |hall|
```

Comment:
There is no need to introduce instance variables here - everything you use is scoped to the method.